### PR TITLE
Supress 1579 when the type being enumerated is an Error Type

### PR DESCRIPTION
**Pull request by YairHalberstadt**
A fix to https://github.com/dotnet/roslyn/issues/32334

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs [hunk 1]
```diff
@@ -3399,5 +3399,93 @@ public static void Main()
...
+class C
+{
+    void Goo()
+    {
+
+        var nonsense = Nonsense;
+        //Should not have error
+        foreach(var nons in nonsense) {}
+
+        var nonsense = new Nonsense();
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs]: You probably want to select a fresh name for this local.

**Comment by gafter** [comment]
You may have to fix the broken test `IForEachLoopStatement_QueryExpression`

**Comment by YairHalberstadt** [comment]
Thanks for spotting that @gafter.
They should both now be done.

**Comment by YairHalberstadt** [comment]
Is there anything else that needs to be done here?
Thanks

File: src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs [hunk 2]
```diff
@@ -623,6 +623,9 @@ private bool GetEnumeratorInfo(ref ForEachEnumeratorInfo.Builder builder, BoundE
             bool wrongAsync = GetEnumeratorInfo(ref ignoredBuilder, collectionExpr, !isAsync, ignoredDiagnostics) == EnumeratorResult.Succeeded;
             ignoredDiagnostics.Free();
 
+            if (collectionExpr.Type.IsErrorType())
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs]: Nit: could just use `collectionExprType` here.

Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs]: >if (collectionExpr.Type.IsErrorType()) [](start = 12, length = 38)

Can this check be moved ahead of retry case above?

Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs]: >collectionExpr.Type [](start = 16, length = 19)

`collectionExprType`

File: src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs [hunk 3]
```diff
@@ -3399,5 +3399,90 @@ public static void Main()
 
             CompileAndVerify(comp, expectedOutput: "1");
         }
+
+        [WorkItem(32334, "https://github.com/dotnet/roslyn/issues/32334")]
+        [Fact]
+        public void SupressForEachMissingMemberErrorOnErrorType()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ForEachTests.cs]: >Supress [](start = 20, length = 7)

`Suppress`

**Comment by YairHalberstadt** [comment]
Thanks
Should all be done now.

**Comment by YairHalberstadt** [comment]
Thanks @gafter. I enjoy working on roslyn!

**Pull request merged**